bench_method.go

```diff
@@ -63,19 +64,35 @@ func (m benchmarkMethod) call(t transport.Transport) (time.Duration, error) {
 	return duration, err
 }
 
+func hostPortBalancer(hostPorts []string) func(i int) string {
```

Comment:
Is this really specific to hostports? 

You might consider moving it to a `balancer.go` file and renaming to `Balance(strs []string) func(i int) string`.